proxy-node-shared/src/main/java/uk/gov/ida/notification/shared/logging/ProxyNodeLoggingFilter.java

```diff
@@ -29,25 +30,24 @@
 
     @Override
     public void filter(ContainerRequestContext requestContext) {
-        String istioTraceId = requestContext.getHeaderString(X_B3_TRACEID);
-        if (istioTraceId != null && !istioTraceId.isBlank()) {
-            MDC.put(X_B3_TRACEID, istioTraceId);
-        }
+        var istioTraceId = ofNullable(requestContext.getHeaderString(X_B3_TRACEID));
+        istioTraceId.filter(Predicate.not(String::isBlank)).ifPresent(traceId -> MDC.put(X_B3_TRACEID, traceId));
```

Comment:
😄

Reply:
What??

Reply:
Actually it's quite clever but I would just chain it instead of splitting into two statements

Reply:
```
ofNullable(requestContext.getHeaderString(X_B3_TRACEID).filter(Predicate.not(String::isBlank)).ifPresent(traceId -> MDC.put(X_B3_TRACEID, traceId));
```

Reply:
Or it might even fit on one line

Reply:
You are right, i thought the variable name istioTraceId had some significance. thanks